client-side-js/injectUI5.js

```diff
@@ -236,6 +236,71 @@ async function clientSide_injectUI5(config, waitForUI5Timeout) {
                         return controlMethodsToProxy
                     }
 
+                    window.wdi5.isCyclic = (obj) => {
+                        var seenObjects = []
+
+                        function detect(obj) {
+                            if (obj && typeof obj === "object") {
+                                if (seenObjects.indexOf(obj) !== -1) {
+                                    return true
+                                }
+                                seenObjects.push(obj)
+                                for (var key in obj) {
+                                    if (obj.hasOwnProperty(key) && detect(obj[key])) {
+                                        console.log(obj, "cycle at " + key)
+                                        return true
+                                    }
+                                }
+                            }
+                            return false
+                        }
+
+                        return detect(obj)
+                    }
+
+                    window.wdi5.removeCyclic = (obj) => {
+                        var seenObjects = []
+
+                        function detect(obj) {
+                            if (obj && typeof obj === "object") {
+                                if (seenObjects.indexOf(obj) !== -1) {
+                                    return obj
+                                }
+                                seenObjects.push(obj)
+                                for (var key in obj) {
+                                    if (obj.hasOwnProperty(key) && detect(obj[key])) {
+                                        console.log(obj, "cycle at " + key)
+                                        console.warn(`deleted: ${key}`)
+                                        delete obj[key]
+                                        return obj
+                                    }
+                                }
+                            } else {
+                                console.log(`removed ${typeof obj}`)
+                            }
+                            return obj
+                        }
+
+                        return detect(obj)
+                    }
+
+                    /**
+                     * JSON.stringify(circularReference, getCircularReplacer());
+                     * @returns
+                     */
+                    window.wdi5.getCircularReplacer = () => {
+                        const seen = new WeakSet()
+                        return (key, value) => {
+                            if (typeof value === "object" && value !== null) {
+                                if (seen.has(value)) {
+                                    return
+                                }
+                                seen.add(value)
+                            }
+                            return value
+                        }
+                    }
+
```

Comment:
is the next circular-thingie function still necessary/used or can its' usage be taken over by any of the above 3?